src/consensus/pbft_manager.cpp

```diff
@@ -915,22 +915,18 @@ void PbftManager::certifyBlock_() {
 
       if (executed_soft_voted_block_for_this_round || unverified_soft_vote_block_for_this_round_is_valid) {
         // generate cert vote
+        // comparePbftBlockScheduleWithDAGblocks_ has checked the cert voted block exist
+        auto cert_voted_block = pbft_chain_->getUnverifiedPbftBlock(soft_voted_block_for_this_round_.first);
+        auto batch = db_->createWriteBatch();
+        db_->addPbftCertVotedBlockHashToBatch(round, soft_voted_block_for_this_round_.first, batch);
+        db_->addPbftCertVotedBlockToBatch(*cert_voted_block, batch);
+        db_->commitWriteBatch(batch);
+        cert_voted_values_for_round_[round] = soft_voted_block_for_this_round_.first;
+        should_have_cert_voted_in_this_round_ = true;
```

Comment:
I don't think it correct. If failed in vote sortition, the node will think it has voted already. Certify state is pulling state, then will not be able to vote again

Reply:
Good points.